python/pyarrow/_parquet.pyx

```diff
@@ -477,23 +428,24 @@ cdef class ParquetSchema:
         if i < 0 or i >= len(self):
             raise IndexError('{0} out of bounds'.format(i))
 
-        cdef ColumnSchema col = ColumnSchema()
-        col.init_from_schema(self, i)
-        return col
+        return ColumnSchema(self, i)
 
 
 cdef class ColumnSchema:
     cdef:
         ParquetSchema parent
         const ColumnDescriptor* descr
 
-    def __cinit__(self):
-        self.descr = NULL
-
-    cdef init_from_schema(self, ParquetSchema schema, int i):
+    def __cinit__(self, ParquetSchema schema, int i):
         self.parent = schema
         self.descr = schema.schema.Column(i)
 
+    def __eq__(self, other):
```

Comment:
How about `__ne__`? Is it generated automatically by Cython?

Reply:
Python's default implementation is for `__ne__` is `not __eq__`

Reply:
That seems true on Python 3, but not on Python 2:
```python
Python 2.7.12 (default, Dec  4 2017, 14:50:18) 
[GCC 5.4.0 20160609] on linux2
Type "help", "copyright", "credits" or "license" for more information.
>>> class C(object):
...   def __eq__(self, other): return True
... 
>>> a = C()
>>> b = C()
>>> a == b
True
>>> a != b
True
>>> 
```

Reply:
```py
Python 3.6.5 |Anaconda, Inc.| (default, Mar 29 2018, 13:14:23)
Type 'copyright', 'credits' or 'license' for more information
IPython 6.4.0 -- An enhanced Interactive Python. Type '?' for help.

In [1]: class E:
   ...:     def __eq__(self, other):
   ...:         return True
   ...:

In [5]: a = E()

In [6]: b = E()

In [7]: a == b
Out[7]: True

In [8]: a != b
Out[8]: False
```

I guess this is a python 2/3 incompatibility. Cython does handle this one, right?

Reply:
I don't know, best would be to add a test to exercise that :-)

Reply:
[Here](https://github.com/apache/arrow/pull/2187/files#diff-348d66d69aec7aedf0a6f753613acb43R677) is a test for `ColumnSchema`